api/controllers/app/install/controller.go

```diff
@@ -78,21 +118,64 @@ func NewInstallController(opts ...InstallControllerOption) (*InstallController,
 		return nil, err
 	}
 
+	if controller.appConfigManager == nil {
```

Comment:
why initialize these if they are nil? how come we werent before? would it be better to return an error?

Reply:
ok